pkg/api/api.go

```diff
@@ -51,14 +52,18 @@ func New(serverIDFunc func() string) *API {
 // ServerID returns the Server ID to be used in responses. A custom server ID
 // generation function can be passed to New().
 func (a API) ServerID() string {
+	serverID := "<unknown>"
 	if a.serverIDFunc != nil {
-		return a.serverIDFunc()
+		serverID = a.serverIDFunc()
+	} else {
+		if hn, err := os.Hostname(); err == nil {
+			serverID = hn
+		}
 	}
-	hn, err := os.Hostname()
-	if err != nil {
-		return "<unknown>"
+	if err := limits.Validator.ValidateServerID(serverID); err != nil {
+		panic(err)
```

Comment:
I think that getting panic with explicit explanation is a better user experience than unpredictable behaviour after silent truncation of input.:) I would prefer to make the check in the API.New function, set value in API struct and return it on demand, but it's another "shared state" :) Also it's not clear if serverIDFunc could return different values. (And if it cannot, why it's a function? :)

Reply:
just to be clear: I am not against state (it's obviously necessary in many cases), I am wary of unnecessary global state, and this specific issue is not about global state

Reply:
`serverIDFunc` is there to allow for different implementations - I believe we use it internally to return a custom name instead of the host name (but I'm going on memory and might be wrong)

Reply:
> I think that getting panic with explicit explanation is a better user experience than unpredictable behaviour after silent truncation of input.:)

That's indeed a better user experience (and the panic is also captured and reported in the status), but what I think both @xaionaro and I are trying to say here is that we would rather have this error handled explicitly, and reported directly via error checking rather than recovering from a `panic` in the job runner

Reply:
Changed API.New to return error instead of panic.